src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4607,8 +4696,75 @@ def update_disk_encryption_set(cmd, instance, client, resource_group_name, key_u
     if enable_auto_key_rotation is not None:
         instance.rotation_to_latest_key_version_enabled = enable_auto_key_rotation
 
+    if federatedClientId is not None:
+        if not hasattr(instance, 'federated_client_id'):
+            raise CLIError('Multi-tenant application client id in disk encryption set is not supported')
```

Comment:
Please use specific error type instead of `CLIError`

Reply:
Could you explain why there is such logic?

Reply:
> Could you explain why there is such logic?

If the sdk version and CLI version will be bound for upgrade, this logic can be removed.